src/coreclr/jit/emitwasm.cpp

```diff
@@ -186,21 +252,70 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
 
     switch (insFmt)
     {
+        case IF_NONE:
+            break;
         case IF_OPCODE:
             dst += emitOutputByte(dst, opcode);
             break;
         case IF_BLOCK:
             dst += emitOutputByte(dst, opcode);
-            dst += emitOutputByte(dst, 0x40);
+            dst += emitOutputByte(dst, 0x40 /* block type of void */);
             break;
         case IF_ULEB128:
+        {
             dst += emitOutputByte(dst, opcode);
-            // TODO-WASM: emit uleb128
+            cnsval_ssize_t constant = emitGetInsSC(id);
+            dst += emitOutputULEB128(dst, (uint64_t)constant);
             break;
+        }
+        case IF_SLEB128:
+        {
+            dst += emitOutputByte(dst, opcode);
+            cnsval_ssize_t constant = emitGetInsSC(id);
+            dst += emitOutputSLEB128(dst, (int64_t)constant);
+            break;
+        }
+        case IF_F32:
+        {
+            dst += emitOutputByte(dst, opcode);
+            // Reinterpret the bits as a double constant and then truncate it to f32,
+            //  then finally copy the raw truncated f32 bits to the output.
+            cnsval_ssize_t bits = emitGetInsSC(id);
+            double         value;
+            float          truncated;
+            memcpy(&value, &bits, sizeof(double));
+            truncated = FloatingPointUtils::convertToSingle(value);
+            memcpy(dst + writeableOffset, &truncated, sizeof(float));
+            dst += sizeof(float);
+            break;
+        }
+        case IF_F64:
+        {
+            dst += emitOutputByte(dst, opcode);
+            // The int64 bits are actually a double constant we can copy directly
+            //  to the output stream.
+            cnsval_ssize_t bits = emitGetInsSC(id);
+            memcpy(dst + writeableOffset, &bits, sizeof(cnsval_ssize_t));
```

Comment:
Other emitters largely avoid using `writeableOffset` and go through the abstractions of `emitOutputByte/Etc`. I think that's a good precedent to follow, and we can do that here by introducing e. g. `emitOutputInt64` (in common code, besides the other similar methods) that'd write the 64 bit value.